source-feed-accumulator.py

```diff
@@ -39,14 +39,20 @@ def accumulate_articles(articles, lang_region):
 
     pathlib.Path(config.OUTPUT_DIR).mkdir(parents=True, exist_ok=True)
 
-    if not config.NO_DOWNLOAD:
-        download_file(feed_file, config.PUB_S3_BUCKET, f"brave-today/{feed_file}")
-        download_file(config.OUTPUT_DIR + config.ARTICLE_HISTORY_FILE.format(LANG_REGION=lang_region),
-                      config.PUB_S3_BUCKET,
-                      f"source-suggestions/{config.ARTICLE_HISTORY_FILE.format(LANG_REGION=lang_region)}")
-
-    with open(feed_file) as feeds:
-        feeds_data = json.loads(feeds.read())
+    try:
```

Comment:
know it's not a big difference, but why not adding the try inside the if clause?

Reply:
The problematic things is opening the file statement, what if we don’t  have file, it will exit all the process.